trie/iterator_test.go

```diff
@@ -99,9 +99,9 @@ func TestNodeIteratorCoverage(t *testing.T) {
 
 	// Gather all the node hashes found by the iterator
 	hashes := make(map[common.Hash]struct{})
-	for it := NewNodeIterator(trie); it.Next(); {
-		if it.Hash != (common.Hash{}) {
-			hashes[it.Hash] = struct{}{}
+	for it := NewNodeIterator(trie); it.Next(true); {
+		if it.Hash() != (common.Hash{}) {
+			hashes[it.Hash()] = struct{}{}
```

Comment:
It's fine here since it's only a test, but live code should cache the result of `.Hash()` since it allocates a new `common.Hash` memory area. Just saying to make sure any code after this will be ok :)